src/backend_model_instance.h

```diff
@@ -62,28 +62,24 @@ class TritonModelInstance {
    public:
     Signature(
         const inference::ModelInstanceGroup& group_config, int32_t device_id)
-        : group_config_(group_config), device_id_(device_id), can_match_(true)
+        : group_config_(group_config), device_id_(device_id)
     {
     }
-    // Check if the lhs signature is equivalent to the rhs, if matching is
-    // enabled. If matching is disabled, lhs != rhs under all scenarios.
     bool operator==(const Signature& rhs) const
     {
-      return can_match_ && rhs.can_match_ && device_id_ == rhs.device_id_ &&
+      return device_id_ == rhs.device_id_ &&
              EquivalentInInstanceConfig(group_config_, rhs.group_config_);
     }
     bool operator!=(const Signature& rhs) const { return !(*this == rhs); }
-    // Enable/Disable matching. If disabled, on either lhs or rhs or both, then
-    // lhs != rhs under all scenarios, including if they are equivalent.
-    // This feature is intended to filter out signatures that have already been
-    // matched, by disabling matching.
-    void EnableMatching() { can_match_ = true; }
-    void DisableMatching() { can_match_ = false; }
+    bool operator<(const Signature& rhs) const
+    {
```

Comment:
how about this instead:

1) use an unordered map
2) define the hash for a signature as the hash of the serialized config (as string)
3) create the hash once when signature is constructed

Unordered map would have better perf - and we don't need an ordered map. creating the hash based on the serialized string (and storing it) - will be more efficient than serializing on each compare -